kotlinpoet/src/main/java/com/squareup/kotlinpoet/CodeWriter.kt

```diff
@@ -634,6 +634,10 @@ internal class CodeWriter constructor(
     modifiers: Set<KModifier>,
     implicitModifiers: Set<KModifier>,
   ): Boolean {
+    if (modifiers.contains(KModifier.OVERRIDE)) {
```

Comment:
I've got two questions here:

1. Is it always the case that we can omit `public` if `override` is present?
2. Do we need to handle `internal` as well, i.e. not emitting the `internal` modifier on overridden `internal` methods?

Reply:
Thanks for reminding me of that! I'll fix it later. These are the only scenarios I can think of right now. Let me know if you have anything else that isn't listed here!

```kotlin
@file:Suppress("ClassName")

package me.omico.kotlin.inheritance.test

public interface Scope {
    // "public" only
    public val id: String
}

public abstract class AbstractScope : Scope {
    // Can omit "public"
    // The bytecode is the same when without "public"
    // But IntelliJ won't prompt to omit
    // Do not omit "final"
    // Cannot make it "protected" or "private"
    override val id: String = "AbstractScope"

    internal abstract val internalProperty: String
    internal abstract fun internalFunction()

    protected abstract val protectedProperty: String
    protected abstract fun protectedFunction()
}

public class ScopeTest1 : AbstractScope() {
    // Can make it "public"
    // Cannot make it "protected" or "private"
    // Same rules for constructor parameters
    public override val internalProperty: String = "ScopeTest1.internalProperty"
    public override fun internalFunction(): Unit = println(internalProperty)

    // Can make it "public" or "protected"
    // Same rules for constructor parameters
    protected override val protectedProperty: String = "ScopeTest1.protectedProperty"
    protected override fun protectedFunction(): Unit = println(protectedProperty)
}

public open class ScopeTest2 : AbstractScope() {
    // Should omit "internal"
    // Same rules for constructor parameters
    internal override val internalProperty: String = "ScopeTest2.internalProperty"
    internal override fun internalFunction(): Unit = println(internalProperty)

    override val protectedProperty: String = "ScopeTest2.protectedProperty"
    override fun protectedFunction(): Unit = println(protectedProperty)
}
```

Reply:
I doubled check on this PR.

1. I can not find a counterexample for it. For now, my answer would be yes; we should always omit `public` if `override` is present.
2. We cannot use `public` and `internal` at the same time. So I don't think we should do anything else here.

Reply:
Wouldn't an explicit `public` on an internal override serve to increase the visibility on that type? Whereas if omitted the member would remain implicitly `internal`.

Reply:
@JakeWharton Yes, you are right! My brain was stuck before. Just submitted a new fix.